include/openenclave/enclave.h

```diff
@@ -641,6 +641,45 @@ oe_result_t oe_generate_attestation_certificate(
     uint8_t** output_cert,
     size_t* output_cert_size);
 
+/**
+ * oe_generate_attestation_certificate_v2
+ *
+ * Similar to oe_generate_attestation_certificate
+ * This function generates a self-signed x.509 certificate with an embedded
+ * quote from the underlying enclave. It also allows a user to pass in
+ * expiration date for the certificate.
+ *
+ * @param[in] subject_name a string contains an X.509 distinguished
+ * name (DN) for customizing the generated certificate. This name is also used
+ * as the issuer name because this is a self-signed certificate
+ * See RFC5280 (https://tools.ietf.org/html/rfc5280) specification for details
+ * Example value "CN=Open Enclave SDK,O=OESDK TLS,C=US"
+ *
+ * @param[in] private_key a private key used to sign this certificate
+ * @param[in] private_key_size The size of the private_key buffer
+ * @param[in] public_key a public key used as the certificate's subject key
+ * @param[in] public_key_size The size of the public_key buffer.
+ * @param[in] date_not_valid_before cert is not valid before this date, format:
+ * YYYYMMDDHHMMSS
+ * @param[in] date_not_valid_after cert is not valid after this date, format:
+ * YYYYMMDDHHMMSS
+ *
+ * @param[out] output_cert a pointer to buffer pointer
+ * @param[out] output_cert_size size of the buffer above
+ *
+ * @return OE_OK on success
+ */
+oe_result_t oe_generate_attestation_certificate_v2(
+    const unsigned char* subject_name,
+    uint8_t* private_key,
+    size_t private_key_size,
+    uint8_t* public_key,
+    size_t public_key_size,
+    const char* date_not_valid_before,
+    const char* date_not_valid_after,
+    uint8_t** output_cert,
+    size_t* output_cert_size);
```

Comment:
```suggestion
    uint8_t** output_certificate,
    size_t* output_certificate_size);
```
https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md#coding-conventions says: "DON'T use abbreviations unless they are already well-known terms known by users (e.g., "app", "info"), or are already required for use by developers (e.g., "min", "max", "args"). Examples of bad use would be num_widgets instead of widget_count, and opt_widgets instead of option_widgets or optional_widgets."